Sources/Apodini/Semantic Model Builder/GraphQL/GraphQLSchemaBuilder.swift

```diff
@@ -0,0 +1,248 @@
+            } else {
+                self.args[leafName, default: [:]][p.name] = GraphQLArgument(type: graphqlType, description: p.description)
+            }
+
+        }
+
+        // Response type and context info
+        let treeTemp = try! EnrichedInfo.node(endpoint.handleReturnType)
+        self.responseTypeTree[leafName] = treeTemp
+        self.leafContext[leafName] = context
+
+        // Handle Single points
+        if (currentPath.count == 1) {
+            self.fields[leafName] = self.graphQLFieldCreator(for: self.responseTypeTree[leafName]!,
+                    self.leafContext[leafName]!,
+                    self.args[leafName] ?? [:])
+            return
+        }
+
+        // Create tree
+        var indx = currentPath.count - 1
+        while (indx >= 1) {
+            let child = currentPath[indx], parent = currentPath[indx - 1]
+            if (self.tree.keys.contains(parent)) {
+                self.tree[parent]!.insert(child)
+            } else {
+                self.tree[parent] = [child]
+            }
+            hasIncomingEdge.insert(child)
+            indx -= 1
+        }
+
+
+    }
+
+    private func nameExtractor(for node: String) -> String {
+        node.components(separatedBy: "_").filter({ $0 != "" }).last ?? "None"
+    }
+
+    // To handle `Names must match /^[_a-zA-Z][_a-zA-Z0-9]*$/`
+    private func graphQLRegexCheck(for str: String) -> String {
+        if (Array(str)[0].isNumber) {
+            return "n_" + str
+        } else {
+            return str
+        }
+    }
+
+    private func generateSchemaFromTreeHelper(_ node: String) -> GraphQLField {
+        let nodeName = self.nameExtractor(for: node)
+        if let childrenList = self.tree[node] {
+            var currentFields = [String: GraphQLField]()
+            if let responseType = self.responseTypeTree[nodeName], let responseContext = self.leafContext[nodeName] { // It has handler
+                let fieldName = self.graphQLRegexCheck(for: responseType.value.typeInfo.name.lowercased())
+                currentFields[fieldName] = self.graphQLFieldCreator(for: responseType,
+                        responseContext,
+                        self.args[nodeName] ?? [:])
+            }
+
+            for child in childrenList {
+                let childName = self.graphQLRegexCheck(for: self.nameExtractor(for: child)) // child.components(separatedBy: "_").filter({ $0 != "" }).last ?? "None"
+                // if let childrenList = self.tree[node] {
+                currentFields[childName] = generateSchemaFromTreeHelper(child)
+            }
+
+            let checkedNodeName = self.graphQLRegexCheck(for: nodeName)
+            self.types[checkedNodeName] = try! GraphQLObjectType(name: checkedNodeName, fields: currentFields)
+            return GraphQLField(type: self.types[checkedNodeName]!, resolve: { _, _, _, _ in "Emtpy" })
```

Comment:
I'm a bit lost. When does this happen? And why are you returning "Empty"?

Reply:
If they are in the middle of the tree